PLT-106: Cluster group resource support. (#181)
spectrocloud/terraform-provider-spectrocloud: examples/e2e/cluster_group/README.md

> # Virtual Cluster demo

Reviewer: ```suggestion
# Virtual Cluster Demo
```

> End-to-end example of provisioning a Palette Virtual Cluster with all of its dependencies. This terraform configuration will provision the following resources on Spectro Cloud:

Reviewer: ```suggestion
This is an end-to-end example of provisioning a Palette Virtual Cluster with all of its dependencies. This Terraform configuration will provision the following resources on Spectro Cloud:
```

- K8s host cluster and datasource.
- Addon Cluster Profile

> ## Instructions:

Reviewer: ```suggestion
## Instructions
```
Removing the colon to keep it the same for all headings

> Clone this repository to a local directory, and then change directory to `examples/e2e/virtual`. Proceed with the following:

Reviewer: ```suggestion
Clone this repository to a local directory, and then change the directory to `examples/e2e/virtual`. Proceed with the following steps:
```

> 1. Provision host cluster as a prerequisite to use virtual cluster. 

Reviewer: ```suggestion
1. Provision a host cluster. This is a prerequisite before using a Palette Virtual Cluster. 
```

> To achieve it add host configuration block to existing cluster or provision it from scratch using one of existing examples:

Reviewer: ```suggestion
To create a virtual cluster, either add a host configuration block to an existing host cluster or provision a new host cluster from scratch using one of the existing examples:
```

<pre>
  host_config {
    host_endpoint_type = "LoadBalancer" 
    ingress_host       = "*.dev.spectrocloud.com"
  }
</pre>

> 2. From the current directory, copy the template variable file `terraform.template.tfvars` to a new file `terraform.tfvars`.

Reviewer: ```suggestion
2. From the current directory, copy the template variable file `terraform.template.tfvars` to a new file with the name `terraform.tfvars`.
```

3. Specify and update all the placeholder values in the `terraform.tfvars` file.

> 4. Initialize and run terraform: `terraform init && terraform apply`.

Reviewer: ```suggestion
4. Initialize Terraform and invoke the deployment with the following command: `terraform init && terraform apply --auto-approve`.
```
@nikchern Not sure if you want us to include the `--auto-approve` flag but otherwise we should mention the prompt.

Author: Will include auto approve.

5. Wait for the cluster creation to finish.

> Once the cluster is provisioned, the cluster _kubeconfig_ file is exported in the current working directly.

Reviewer: ```suggestion
Once the cluster is provisioned, the cluster's _kubeconfig_ file is exported directly into the current working directory.
```

> Export the kubeconfig and check cluster pods:

Reviewer: ```suggestion
Export the kubeconfig and check the cluster pods:
```

```shell
export KUBECONFIG=kubeconfig_eks
kubectl get pod -A
```

> ## Cleanup:

Reviewer: ```suggestion
## Clean Up
```
Clean up is two words when it's an action (verb), but if you are talking about the act of cleaning such as "the cleanup yesterday took forever" then it's one word 😄

Reviewer: @nikchern this one got missed.
